csrc/scheduler/matmul.cpp

```diff
@@ -614,47 +617,96 @@ void scheduleOutputTensor(
       {c},
       {ParallelType::BIDx, ParallelType::BIDy, ParallelType::BIDz});
 }
+
+//! Assign parallelization modes to epilogue inputs. The returned pair of
+//!  collections contains:
+//!   - 1st: TVs that can have all parallelization modes propagated,
+//!   - 2nd: TVs that can have all but vectorization parallel modes
+//!     propagated,
+std::pair<std::vector<TensorView*>, std::vector<TensorView*>>
+partitionEpilogueInputsByParallelizationModes(
+    TensorView* reference_tv,
+    const std::vector<TensorView*>& epilogue_inputs) {
+  const auto& tvs_with_vec = scheduler_utils::getInputsOutputsWithInnerDim(
```

Comment:
Is this a reference to temporary object? `getInputsOutputsWithInnerDim` returns `std::vector<TensorView*>`, instead of returning a reference.